### ISSUE TITLE
Reduce test matrix to newest and oldest python version

### ISSUE THREAD

**Issue by dkrako** [pymovements/pymovements#389]
## Description of the problem
Our test matrix currently spans 3 x 5 environments (3 OS, 5 python versions) which seems to be a bit wasteful on resources.

Should we really check each intermediate python version from 3.7 to 3.11 or isn't it sufficient to test only on 3.7 and 3.11?

**Comment by dkrako** [comment]
@cbueth @SiQube What do you think about reducing the PR test matrix to newest and oldest?

However I think we should still test the main branch on all pymovements versions. Although this should be very rare, it might be that a PR passes despite the merged main commit fails on a specific python version, so I'm not sure if this is a good decision.

Although in most day-to-day development this doesn't make much of an impact, the bottleneck was very noticeable during our workshops where lots of people develop simultaneously.

**Comment by cbueth** [comment]
Agreed! I think we should lift off some weight of the PR CIs. A middle ground would be having only one OS test all versions, but with continuous development just testing min and max must be enough, even for main. I'd vote for min&max for PRs to `main`.

**Comment by SiQube** [comment]
sounds very reasonable
